src/DataTransferObject.php

```diff
@@ -82,10 +82,40 @@ public function except(string ...$keys): DataTransferObject
     public function toArray(): array
     {
         if (count($this->onlyKeys)) {
-            return Arr::only($this->all(), $this->onlyKeys);
+            $array = Arr::only($this->all(), $this->onlyKeys);
+        } else {
+            $array = Arr::except($this->all(), $this->exceptKeys);
         }
 
-        return Arr::except($this->all(), $this->exceptKeys);
+        foreach ($array as $key => $property) {
+            $array[$key] = $this->convertForArray($property);
+        }
+
+        return $array;
+    }
+
+    /**
+     * @param mixed $property
+     *
+     * @return mixed
+     */
+    protected function convertForArray($property)
```

Comment:
I'm not 100% certain about the name here. Maybe `convertPropertyToArray`? Feel free to share your opinion!

Reply:
I'm agree. Naming is a pain) Your suggest is better.